Develop

efrapp opened the pull request.

efrapp committed:
Update README.md

efrapp committed:
Create User model

efrapp committed:
Merge branch 'develop' of github.com:efrapp/members_only into develop

efrapp committed:
- Sign In form added
- User model validations added
- Session controller created to manage login process

efrapp committed:
Merge branch 'master' into develop

efrapp committed:
Apply TDD for signup process

efrapp committed:
- Genereate remember token callback created
- test for sign in users added
- sign in method created to set session for user id and cookie for remember token

efrapp committed:
- Create current user functionality
- adding test to verify that remember token changes when user sign in
- create sign out functionality
- create test for sign out functionality

efrapp committed:
Post creation
- create view for creation post form
- create action controller to create post in databe
- create test for post with valid and invalid data
- move method in ApplicationController to SessionsHelper to use them on tests

efrapp committed:
Show list of posts
- Create index action to show posts order by most recent
- Create tests to validate that only signed in user can see author of posts
- Fixing style for header navbar
- Redirect to user profile when it tries to check signup page

efrapp committed:
Redirect to user profile when it tries to check signin page

rvvergara commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,38 @@
+    @user = current_user
+  end
+
+  def create
```
Once the user is created, it would be better for the user to be signed in directly. At the moment, even upon signup the user still needs logging in to do anything.

efrapp replied:
Hi Ryan,
Ok, I made the change but I want you to confirm to me that the "All Posts" link works even if there isn't an account created because, on my end, I can see all the posts. Thank you!

rvvergara commented on app/models/post.rb:
```diff
@@ -0,0 +1,8 @@
+  validates :title, presence: true
+  validates :body, presence: true
+
+  scope :order_by_recent, -> { order(created_at: :desc) }
```
Can we just make this as the default_scope? Anyway there is only one way we show the posts.

rvvergara commented on app/controllers/posts_controller.rb:
```diff
@@ -0,0 +1,47 @@
+  end
+
+  def index
+    @posts = Post.all.order_by_recent
```
If we do a default_scope as mentioned in my comment in post.rb then we can simply do a `Post.all` in here

rvvergara commented on config/routes.rb:
```diff
@@ -1,3 +1,13 @@
 Rails.application.routes.draw do
+  root 'users#new'
+  get 'users/new'
```
Since we already have `:users` as a resource, we might not need this line anymore

rvvergara commented on test/models/user_test.rb:
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Though not required for this project but since you requested checking for the tests, I wonder why you left off testing the models?

efrapp replied:
Yeah, you are right. I'm going to apply some tests for them. Thank you.

rvvergara reviewed (changes requested):
Hi @efrapp,

Great job on this! The User Interface is nice and all features are working nicely. There are just some minor stuff I hope you can work on. For instance it would be great if the user automatically gets signed in after account creation. As it is right now, a new user still needs to log in even after he/she has just created his/her account. 

I looked into the tests and yeah you did good on them too. I just think you should also write tests for the models (i.e, validations and associations testing are important). And for future projects I would recommend using RSPec over Minitest. RSpec has a much simpler syntax in my perspective. And also practice testing for the views and user interaction with the pages. 

Just work on the minor changes I requested and we'll see after another round of reviews.

efrapp committed:
First Code Review changes
- log in user directly after sign up
- change scope to default scope to show most recent posts first
- add testing for user and post models
- removing unneeded new user path in router file

rvvergara reviewed (changes requested):
Hi @efrapp ,

I can confirm that everything works fine now! All Posts link works as expected (signed in users see authors while guests don't). And yeah now when I sign up i am automatically logged on. 

Ok great job on this but before I give my thumbs up there's one last thing I want you to work at. Your README. It's currently still the default README provided by Rails. Please provide useful information in your README. You may use [this](https://github.com/RichardLitt/standard-readme) as a guide.

efrapp committed:
Update README.md

efrapp commented:
Ok @rvvergara I made the changes in the Readme, thanks.

rvvergara commented on README.md:
```diff
@@ -1,24 +1,14 @@
-# README
+# Memebers Only Application
```
# Members Only (typo in the word Memebers)

efrapp replied:
Oh thank you Ryan. I'm going to fix it right away.

rvvergara commented on README.md:
```diff
@@ -1,24 +1,14 @@
-This README would normally document whatever steps are necessary to get the
-application up and running.
+> Application to build an exclusive clubhouse where your members can write embarrassing posts about non-members. Inside the clubhouse, members can see who the author of a post is but, outside, they can only see the story and wonder who wrote it.
 
```
For future projects it would also be nice to have sections here that explain how others can install the project. 

Another is a link to a Live Demo.

efrapp replied:
Ok you are right, I didn't put the installation section because I don't how deep I need to go with the steps but I'm going to put it. About the Live Demo, I didn't put it because I'm almost full with the links on Heroku so I want to save some of them for other projects.

rvvergara reviewed (approved):
Hi @efrapp ,

I think your project is ready to go. I won't tie up your time any longer here but there is a typo in your Readme. Also I provided a note in the same that you should consider for future projects.

For now here's to ya:

# :+1: 

You may now submit a completion form and merge this to master

efrapp committed:
Update README.md

efrapp committed:
Adding Install and Ruby Vesion section to README
